spec/std/http/server/handlers/static_file_handler_spec.cr

```diff
@@ -177,6 +177,183 @@ describe HTTP::StaticFileHandler do
     end
   end
 
+  context "when a Range header is provided" do
+    describe "valid Range header" do
+      it "serves a byte range" do
+        headers = HTTP::Headers{"Range" => "bytes=0-2"}
+        response = handle HTTP::Request.new("GET", "/range.txt", headers)
+
+        response.status_code.should eq(206)
+        response.body.should eq "Hel"
+      end
+
+      it "serves a single byte" do
+        headers = HTTP::Headers{"Range" => "bytes=0-0"}
+        response = handle HTTP::Request.new("GET", "/range.txt", headers)
+
+        response.status_code.should eq(206)
+        response.body.should eq "H"
+      end
+
+      it "serves an open-ended byte range" do
+        headers = HTTP::Headers{"Range" => "bytes=6-"}
+        response = handle HTTP::Request.new("GET", "/range.txt", headers)
+
+        response.status_code.should eq(206)
+        response.body.should eq "world\n"
+      end
+
+      describe "suffix range" do
+        it "partial" do
+          headers = HTTP::Headers{"Range" => "bytes=-6"}
+
+          response = handle HTTP::Request.new("GET", "/range.txt", headers)
+
+          response.status_code.should eq(206)
+          response.body.should eq "world\n"
+        end
+
+        it "more bytes than content" do
+          headers = HTTP::Headers{"Range" => "bytes=-15"}
+
+          response = handle HTTP::Request.new("GET", "/range.txt", headers)
+
+          response.status_code.should eq(206)
+          response.body.should eq "Hello world\n"
+        end
+
+        it "zero" do
+          headers = HTTP::Headers{"Range" => "bytes=-0"}
+
+          response = handle HTTP::Request.new("GET", "/range.txt", headers)
+
+          response.status_code.should eq(400)
+        end
+
+        it "negative size" do
+          headers = HTTP::Headers{"Range" => "bytes=--2"}
+
+          response = handle HTTP::Request.new("GET", "/range.txt", headers)
+
+          response.status_code.should eq(400)
+        end
+      end
+
+      it "serves multiple byte ranges (separator without whitespace)" do
+        headers = HTTP::Headers{"Range" => "bytes=0-1,6-7"}
+
+        response = handle HTTP::Request.new("GET", "/range.txt", headers)
+
+        response.status_code.should eq(206)
+        count = 0
+        MIME::Multipart.parse(response) do |headers, part|
+          chunk = part.gets_to_end
+          case range = headers["Content-Range"]
+          when "bytes 0-1/12"
+            chunk.should eq "He"
+          when "bytes 6-7/12"
+            chunk.should eq "wo"
+          else
+            raise "Unknown range: #{range.inspect}"
```

Comment:
This is mostly stylistic, but this is considered a spec failure and should use `fail` instead of `raise`, ditto for below